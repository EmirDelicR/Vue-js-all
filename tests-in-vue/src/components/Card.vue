<template>
    <div class="vte__cool-card">
        <p data-test="card-div" v-text="'hello I am a card :)'" />
        <p>Current Total:</p>
        <p>{{ currentTotal }}</p>
        <button @click="incrementTotal">Increment</button>
    </div>
</template>

<script>
import { mapGetters } from 'vuex'
export default {
    name: 'Card',
    data () {
        return {
            foo: '',
        }
    },
    computed: {
        ...mapGetters('Card', {
            currentTotal: 'currentTotal',
            projectId: 'projectId'
        }),
        getText () {
            return `${this.foo} is Sparta`;
        }
    },
    methods: {
        incrementTotal () {
            this.$store.dispatch('Card/incrementTotal', this.currentTotal + 1);
        },
        addNumbers (num1, num2) {
            return num1 + num2;
        }

    }
};
</script>

<style>
.vte__cool-card {
  height: 20vh;
  width: 40vw;
  padding: 2rem;
  border-radius: 1rem;
  box-shadow: 0px 1px 1px #000;
}
</style>