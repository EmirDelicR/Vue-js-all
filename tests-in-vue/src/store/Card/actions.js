const incrementTotal = (context, newTotal) => {
  context.commit('UPDATE_TOTAL', newTotal)
}

const updateProjectId = (context, newProjectId) => {
  context.commit('UPDATE_PROJECT_ID', newProjectId)
}

export default {
  incrementTotal,
  updateProjectId,
}
