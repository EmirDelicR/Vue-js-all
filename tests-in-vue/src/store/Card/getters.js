const projectId = state => state.projectId
const currentTotal = state => state.currentTotal

export default {
  projectId,
  currentTotal,
}
