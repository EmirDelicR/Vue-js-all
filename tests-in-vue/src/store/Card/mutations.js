const UPDATE_PROJECT_ID = (state, projectId) => {
  state.projectId = projectId
}

const UPDATE_TOTAL = (state, total) => {
  state.currentTotal = total
}

export default {
  UPDATE_PROJECT_ID,
  UPDATE_TOTAL,
}
