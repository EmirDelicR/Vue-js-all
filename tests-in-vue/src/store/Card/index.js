import actions from './actions'
import getters from './getters'
import mutations from './mutations'

const defaultState = {
  currentTotal: 0,
  projectId: null,
}

export default {
  namespaced: true,
  state: defaultState,
  actions,
  getters,
  mutations,
}
