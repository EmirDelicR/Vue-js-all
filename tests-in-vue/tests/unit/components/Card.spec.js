import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Card from '@/components/Card.vue'
import StoreFactory from '../../testUtils/factory'

/** Create local vue instance */
const localVue = createLocalVue()
/** Use vuex */
localVue.use(Vuex)

/** Mock incrementTotal function */
const actions = {
  incrementTotal: jest.fn(),
}

describe('Card-Module-Mock-Store-in-Test', () => {
  let store
  beforeEach(() => {
    /** Mocking store (one way)*/
    store = new Vuex.Store({
      modules: {
        Card: {
          namespaced: true,
          actions,
          getters: {
            projectId: () => 4,
            currentTotal: () => 789,
          },
        },
      },
    })
  })
  test('card should show correct total', () => {
    const wrapper = mount(Card, { store, localVue })
    expect(wrapper.vm.currentTotal).toEqual(789)
  })
})

describe('Card-Module-Mock-Store-From-Utils', () => {
  let store
  beforeEach(() => {
    store = StoreFactory(
      {
        Card: {
          projectId: 3,
          currentTotal: 567,
        },
      },
      actions,
    )
  })

  /** Make a snapshot for component to match */
  test('should render content correctly', () => {
    const wrapper = mount(Card, { store, localVue })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  /** Testing some method from Card component */
  test('addNumbers should add numbers together', () => {
    const wrapper = mount(Card, { store, localVue })
    expect(wrapper.vm.addNumbers(1, 2)).toEqual(3)
  })

  /** Testing computed property from Card Module */
  test('getText should evaluate foo correctly', () => {
    const wrapper = mount(Card, { store, localVue })
    wrapper.setData({
      foo: 'This',
    })
    expect(wrapper.vm.getText).toEqual('This is Sparta')
  })

  test('card should show correct total', () => {
    const wrapper = mount(Card, { store, localVue })
    expect(wrapper.vm.currentTotal).toEqual(567)
  })

  test('card should show correct project id', () => {
    const wrapper = mount(Card, { store, localVue })
    expect(wrapper.vm.projectId).toEqual(3)
  })
})
