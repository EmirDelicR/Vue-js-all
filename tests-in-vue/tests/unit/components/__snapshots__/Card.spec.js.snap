// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card-Module-Mock-Store-From-Utils should render content correctly 1`] = `
<div
  class="vte__cool-card"
>
  <p
    data-test="card-div"
  >
    hello I am a card :)
  </p>
   
  <p>
    Current Total:
  </p>
   
  <p>
    567
  </p>
   
  <button>
    Increment
  </button>
</div>
`;
